spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationComponentScanRegistrar.java

```diff
@@ -150,7 +154,7 @@ private void filter(BeanDefinitionRegistry registry, Map<String, Object> compone
 	}
 
 	protected Collection<String> getBasePackages(AnnotationMetadata importingClassMetadata,
-			@SuppressWarnings("unused") BeanDefinitionRegistry registry) {
+			BeanDefinitionRegistry registry) {
```

Comment:
? It is still unused (this was added to suppress a Sonar issue).

Reply:
Hm. I'm always super suspicious for any code which IDEA shows as  redundant.
I'm not sure what the problem sees Sonar, but that's fully OK to override the method and don't use all the provided params.
🤷

Reply:
Sounds like a bug in IDEA to me; the variable is clearly not used here; whether the super method uses it is irrelevant.

Reply:
```suggestion
			@SuppressWarnings("unused") BeanDefinitionRegistry registry) {
```